beacon-chain/forkchoice/doubly-linked-tree/store.go

```diff
@@ -74,7 +75,7 @@ func (s *Store) head(ctx context.Context) ([32]byte, error) {
 		if s.justifiedCheckpoint.Epoch == params.BeaconConfig().GenesisEpoch {
 			justifiedNode = s.treeRootNode
 		} else {
-			return [32]byte{}, errUnknownJustifiedRoot
+			return [32]byte{}, errors.WithMessage(errUnknownJustifiedRoot, fmt.Sprintf("%#x", s.justifiedCheckpoint.Root))
```

Comment:
Why did you prefer `WithMessage` over `Wrap` here? The difference is that `Wrap` annotates the error wrapper with a stack trace and a message; `WithMessage` only adds the message, not the stack trace.

If you do want to omit the stack trace, `errors.WithMessagef(errUnknownFinalizedRoot, "%#x", finalizedRoot)` is probably preferable to calling `fmt.Sprintf` inline.

Reply:
I thought about `Wrap` but my naive understanding is there's nothing to wrap here. This is the very first error (i.e. the bottom of the chain)